newdle/client/src/components/AvailabilityRing.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+AvailabilityRing.defaultProps = {
+  strokeWidth: 10,
+  radius: 50,
+};
+
+export default function AvailabilityRing({
+  available,
+  ifNeeded,
+  totalParticipants,
+  strokeWidth,
+  radius,
+}) {
+  let size = radius * 1.1;
+  let stroke = 7;
+  let normalizedRadius = radius - stroke * 2;
+  let circumference = normalizedRadius * 2 * Math.PI;
+
+  const greenOffset = `rotate(-90 ${size} ${size})`;
+  const yellowOffset = `rotate(${-90 + (available / totalParticipants) * 360} ${size} ${size})`;
+  const redOffset = `rotate(${-90 +
+    ((available + ifNeeded) / totalParticipants) * 360} ${size} ${size})`;
+
+  let style = {
+    'width': '25%',
+    'color': 'black',
+    'text-align': 'center',
+    'margin': 'auto',
+  };
+  return (
+    <>
+      <svg height={size * 2} width={size * 2}>
+        <circle
+          stroke="#00ac46" // green - grün
+          fill="transparent"
+          strokeWidth={strokeWidth}
+          strokeDasharray={
+            circumference * (available / totalParticipants) +
+            ' ' +
+            circumference * ((totalParticipants - available) / totalParticipants)
+          }
+          transform={greenOffset}
+          r={normalizedRadius}
+          cx={size}
+          cy={size}
+        />
+        <circle
+          stroke="#fdc500" // yellow - gelb
+          fill="transparent"
+          strokeWidth={strokeWidth}
+          strokeDasharray={circumference * (ifNeeded / totalParticipants) + ' ' + circumference}
+          transform={yellowOffset}
+          r={normalizedRadius}
+          cx={size}
+          cy={size}
+        />
+        <circle
+          stroke="#dc0000" // red - rot
+          fill="transparent"
+          strokeWidth={strokeWidth}
+          strokeDasharray={
+            circumference * ((totalParticipants - (available + ifNeeded)) / totalParticipants) +
+            ' ' +
+            circumference
+          }
+          transform={redOffset}
+          r={normalizedRadius}
+          cx={size}
+          cy={size}
+        />
+      </svg>
+      <div style={style}>{available + '/' + totalParticipants}</div>
```

Comment:
I think having this inside would look nicer - but only if we can properly center it. If not maybe it's better to keep it below.